tests/requests/api/v1/playlists/favorites.spec.js

```diff
@@ -97,4 +92,63 @@ describe('Test the Playlists path', () => {
       expect(res.body.message).toBe("ID must be a number!")
     })
   });
+
+  describe('Test Favorites Index', () => {
+    it('happy path', async () => {
+      let favoriteData_2 = {
+        "title": "We are the Champions",
+        "artistName": "Queen",
+        "genre": "Rock",
+        "rating": 100
+      }
+      let playlist   = await database('playlists').select().first()
+      let favorite_1 = await database('favorites').select().first()
+      let favorite_2 = await database('favorites').insert(favoriteData_2, ['id', 'title', 'artistName', 'rating', 'genre']);
+
+      await database('playlist_favorites').insert({playlistId: playlist.id, favoriteId: favorite_1.id}, 'id')
+      await database('playlist_favorites').insert({playlistId: playlist.id, favoriteId: favorite_2[0].id}, 'id')
+
+      const res = await request(app)
+        .get(`/api/v1/playlists/${playlist.id}/favorites`)
+
+      expect(res.statusCode).toBe(200);
+      expect(res.body).toHaveProperty('id');
+      expect(res.body).toHaveProperty('title');
+      expect(res.body.title).toBe('Coding Vibes')
+      expect(res.body).toHaveProperty('createdAt');
+      expect(res.body).toHaveProperty('updatedAt');
+      expect(res.body).toHaveProperty('songCount');
+      expect(res.body.songCount).toBe(2)
+      expect(res.body).toHaveProperty('songAvgRating');
+      expect(res.body.songAvgRating).toBe(90.5)
+      expect(res.body).toHaveProperty('favorites');
+      expect(res.body.favorites[0].id).toBe(favorite_1.id)
+      expect(res.body.favorites[0].title).toBe(favorite_1.title)
+      expect(res.body.favorites[0].artistName).toBe(favorite_1.artistName)
+      expect(res.body.favorites[0].genre).toBe(favorite_1.genre)
+      expect(res.body.favorites[0].rating).toBe(favorite_1.rating)
+
+      expect(res.body.favorites[1].id).toBe(favorite_2[0].id)
+      expect(res.body.favorites[1].title).toBe(favorite_2[0].title)
+      expect(res.body.favorites[1].artistName).toBe(favorite_2[0].artistName)
+      expect(res.body.favorites[1].genre).toBe(favorite_2[0].genre)
+      expect(res.body.favorites[1].rating).toBe(favorite_2[0].rating)
+    });
```

Comment:
This primarily goes along with the comment above. We need to test for when there are no favorites for a playlist.

Reply:
I added testing for that scenario. Very good sad path suggestion